### PR TITLE
Enable FunctionHookMode tracing  by default

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

- Enable FunctionHookMode tracing by default
- Update examples:
  - torch/ssd300_vgg16: remove disable tracing for normalization (is works correctly but changed final metric 0.5148 vs 0.5145)
  - qat/torch/anomalib: in model added new FQ in result changes graph, on windows changed metric (win:  0.9692, linux: 0.9767) 
- check model devices to insert modules in load_from_config 
- Update lora example: 
   - changes to support new tracing
   - add filter for tracing warning to clean outputs
   - update gitignore to ignore output files

### Related tickets

152996

### Tests

https://github.com/openvinotoolkit/nncf/actions/runs/14352342788/job/40233792090
https://github.com/openvinotoolkit/nncf/actions/runs/14353482882/job/40237562175
NNCF/job/manual/job/post_training_quantization/643/

### CODE REVIEW

**Comment by AlexanderDokuchaev** [comment]
Diff in ssd300:
image

**Comment by AlexanderDokuchaev** [comment]
Diff in anomalib 
image
image

**Comment by alexsu52** [comment]
> Diff in anomalib image image

Is the left operand of the multiply ops quantized?

File: nncf/torch/graph/graph.py [hunk 1]
```diff
@@ -102,16 +106,34 @@ def get_nodes_with_missed_input_edges(self) -> List[NNCFNode]:
...
-                num_expected_input_edges = node.metatype.num_expected_input_edges
-            if node.layer_attributes is not None and isinstance(node.layer_attributes, MultipleInputLayerAttributes):
-                num_expected_input_edges = node.layer_attributes.num_inputs
-            if num_expected_input_edges:
-                input_edges = self.get_input_edges(node)
-                if len(input_edges) < num_expected_input_edges:
-                    # If node has missed input edges we assume this node is an input node
-                    # that was disconnected from an activation input.
+
+        if is_torch_tracing_by_torch_function_mode():
```
Comment (alexsu52) [inline nncf/torch/graph/graph.py:110]: To solve this problem, you can use inheritance.

File: nncf/experimental/common/check_feature.py [hunk 2]
```diff
@@ -12,10 +12,13 @@
...
+    Checks if legacy torch tracing is enabled by environment variable NNCF_TORCH_LEGACY_TRACING.
 
-    :return: True if experimental torch tracing is enabled, False otherwise.
+    True - will wrap model by NNCFNetwork and patch function in torch namespace.
+    False - will use FunctionHookMode without patching torch namespace.
+
+    :return: True if legacy torch tracing is enabled, False otherwise.
     """
-    return os.getenv("NNCF_EXPERIMENTAL_TORCH_TRACING") is not None
+    return os.getenv("NNCF_TORCH_LEGACY_TRACING", "").lower() not in ["1", "on", "true"]
```
Comment (alexsu52) [inline nncf/experimental/common/check_feature.py:24]: The user can call patch_torch_operators in their code to enable the tracing via patching. NNCF should handle this for legacy algorithms. At least, `is_torch_tracing_by_torch_function_mode` should handle the case when the tracing via patching is enabled. Regarding the `NNCF_TORCH_LEGACY_TRACING` environment variable, can you provide a user scenario how an user would use it? If there are no custom scripts I would suggest deleting this.

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: examples/post_training_quantization/torch/ssd300_vgg16/main.py [hunk 3]
```diff
@@ -141,7 +140,6 @@ def main():
     model.eval()
 
     # Disable NNCF tracing for some methods in order for the model to be properly traced by NNCF
-    disable_tracing(GeneralizedRCNNTransform.normalize)
```
[the remark this answers is not in the document]
Comment (AlexanderDokuchaev) [reply examples/post_training_quantization/torch/ssd300_vgg16/main.py:144]: because it was added to avoid tracing error, now it works correctly.
Other `disabel_tracing` can be removed too but in this case propagation solver works too long time (too many nodes in the graph)

File: nncf/common/factory.py [hunk 4]
```diff
@@ -90,13 +90,13 @@ def create(model: TModel, inplace: bool = False) -> ModelTransformer[Any]:
...
             return OVModelTransformer(cast(Model, model), inplace=inplace)
-        if model_backend == BackendType.TORCH and is_experimental_torch_tracing_enabled():
+        if model_backend == BackendType.TORCH and is_torch_tracing_by_torch_function_mode():
             from nncf.experimental.torch2.function_hook.nncf_graph.nncf_graph_builder import GraphModelWrapper
             from nncf.experimental.torch2.model_transformer import PT2ModelTransformer
 
             return PT2ModelTransformer(cast(GraphModelWrapper, model))
 
-        if model_backend == BackendType.TORCH and not is_experimental_torch_tracing_enabled():
+        if model_backend == BackendType.TORCH and not is_torch_tracing_by_torch_function_mode():
```
[the remark this answers is not in the document]
Comment (AlexanderDokuchaev) [reply nncf/common/factory.py:99]: i think if we keep old tracing, is will be good to have option to enable it.
This PR only about enable new tracing by default, not about removing any features.

[the remark this answers is not in the document]
Comment (alexsu52) [reply nncf/common/factory.py:99]: ok

**Review by alexsu52 — COMMENTED** [review]
Please check your PR with https://github.com/openvinotoolkit/nncf/pull/3322 and run the nightly training test for Torch backend.
cc' @ljaljushkin

File: nncf/experimental/torch2/function_hook/serialization.py [hunk 5]
```diff
@@ -101,11 +104,19 @@ def load_from_config(model: TModel, config: Dict[str, Any]) -> TModel:
...
+        device = get_model_device(wrapped_model)
+    else:
+        nncf_logger.warning("Model is on multiple devices. Cannot determine device for loaded modules.")
+
     hook_storage = get_hook_storage(wrapped_model)
     transformation_commands = cast(List[S_COMMAND], config[COMPRESSION_STATE_ATTR])
     for command in transformation_commands:
         module_cls = COMPRESSION_MODULES.get(command["module_cls_name"])
         module = module_cls.from_config(command["module_config"])
+        module.to(device)
```
Comment (ljaljushkin) [inline nncf/experimental/torch2/function_hook/serialization.py:119]: I encountered the same problem with the model on various devices:
https://github.com/openvinotoolkit/nncf/pull/3341/files#diff-f257219eb5631518ca7ec37bb9858c77491049dc999ca14d845a46517f263900R135-R136

Is there a reliable method to match the compression module from the insertion command to the target layer and its parameters? With this matching, it would be possible to determine the device.

Comment (AlexanderDokuchaev) [reply nncf/experimental/torch2/function_hook/serialization.py:119]: Currently we have limited support for multidevice models.
For not multi device model it works.

File: examples/quantization_aware_training/torch/resnet18/main.py [hunk 6]
```diff
@@ -34,6 +34,8 @@
 import nncf
 import nncf.torch
 from nncf.common.utils.helpers import create_table
+from nncf.experimental.torch2.function_hook.serialization import get_config
```
[the remark this answers is not in the document]
Comment (AlexanderDokuchaev) [reply examples/quantization_aware_training/torch/resnet18/main.py]: done

File: nncf/torch/__init__.py [hunk 7]
```diff
@@ -50,6 +50,7 @@
 from nncf.torch.model_creation import is_wrapped_model
 from nncf.torch.model_creation import wrap_model
 from nncf.torch.model_creation import load_from_config
+from nncf.experimental.torch2.function_hook.serialization import get_config
```
Comment (alexsu52) [inline nncf/torch/__init__.py:53]: I would suggest to implement `get_config` function in the model_creation.py to support `NNCF_TORCH_LEGACY_TRACING` in case of the user migrate to `nncf.torch.get_config`. I hope this is not a big deal.

I would suggest to add a telemetry for get_config function.

Comment (AlexanderDokuchaev) [reply nncf/torch/__init__.py:53]: done

File: docs/usage/training_time_compression/quantization_aware_training/Usage.md [hunk 8]
```diff
@@ -96,7 +96,7 @@ resuming_checkpoint = torch.load(path)
 nncf_config = resuming_checkpoint['nncf_config']
 state_dict = resuming_checkpoint['state_dict']
 
-quantized_model = nncf.torch.load_from_config(model, nncf_config, dummy_input)
+quantized_model = nncf.torch.load_from_config(model, nncf_config)
```
[the remark this answers is not in the document]
Comment (AlexanderDokuchaev) [reply docs/usage/training_time_compression/quantization_aware_training/Usage.md:99]: > using nncf.torch.get_config() / nncf.torch.load_from_config() in Training-Time Quantization

 done 

> add note that create_compressed API is deprecated in Training-Time Compression.

we have already have deprecation message in create_compressed sine previous release, what a reason to duplicate it in readme now?

File: nncf/experimental/torch2/function_hook/serialization.py (hunk 5 above)
Comment (AlexanderDokuchaev) [reply nncf/experimental/torch2/function_hook/serialization.py:119]: I did not look at it. Currently multidevice support for new tracing works similar like for current.

File: nncf/torch/model_creation.py [hunk 9]
```diff
@@ -114,6 +116,14 @@ def create_compressed_model(
         " - https://github.com/openvinotoolkit/nncf/tree/develop/examples/quantization_aware_training/torch"
     )
 
+    print(PATCHING_STATE.operators_are_wrapped)
```
[the remark this answers is not in the document]
Comment (AlexanderDokuchaev) [reply nncf/torch/model_creation.py]: removed

**Review by alexsu52 — APPROVED** [review]

**Review by ljaljushkin — APPROVED** [review]
